pkg/controller/kieapp/shared/utils.go

```diff
@@ -174,3 +180,51 @@ func GetNamespacedName(object metav1.Object) types.NamespacedName {
 		Namespace: object.GetNamespace(),
 	}
 }
+
+func GetConsoleYAMLSample(res resource.KubernetesResource) (*consolev1.ConsoleYAMLSample, error) {
+	snippetStr := res.GetAnnotations()["consoleSnippet"]
+	var snippet bool = false
+	if snippetStr != "" {
+		tmp, err := strconv.ParseBool(snippetStr)
+		if err != nil {
+			return nil, errors.New("Unable to parse snippet to boolean.")
+		}
+		snippet = tmp
+	}
+
+	title, _ := res.GetAnnotations()["consoleTitle"]
+	desc, _ := res.GetAnnotations()["consoleDesc"]
+	name, _ := res.GetAnnotations()["consoleName"]
+	if name == "" {
+		name = res.GetName() + "-yamlsample"
+	}
+
+	curannos := res.GetAnnotations()
+	delete(curannos, "consoleSnippet" )
+	delete(curannos, "consoleTitle" )
+	delete(curannos, "consoleDesc" )
+	delete(curannos, "consoleName" )
+
+	data, err := yaml.Marshal(res)
+	if err != nil {
+		return nil, errors.New("Failed to convert to yamlstr from KubernetesResource.")
+	}
+
+	yamlSample := &consolev1.ConsoleYAMLSample{
+		ObjectMeta: amv1.ObjectMeta{
+			Name:      name,
+			Namespace: "openshift-console",
+		},
+		Spec: consolev1.ConsoleYAMLSampleSpec{
+			TargetResource: amv1.TypeMeta{
+				APIVersion: res.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+				Kind:       res.GetObjectKind().GroupVersionKind().Kind,
+			},
+			Title:       consolev1.ConsoleYAMLSampleTitle(title),
+			Description: consolev1.ConsoleYAMLSampleDescription(desc),
+			YAML:        consolev1.ConsoleYAMLSampleYAML(string(data)),
+			Snippet:     snippet,
+		},
+	}
+	return yamlSample, nil
+}
```

Comment:
I think this looks good otherwise. Please find a good folder and package name for it in operator-utils and open a PR there. Also create a unit test for it there, and let's prepare to merge it.

Reply:
will do.  first, will make this kie-cloud pr working first.